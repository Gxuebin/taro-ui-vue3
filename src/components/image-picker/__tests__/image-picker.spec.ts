import { mount } from '@vue/test-utils'
import AtImagePicker from '../index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtImagePicker as any, {
    slots,
    props: { ...values },
  })
}

describe('AtImagePicker', () => {
  const files = [
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121',
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg',
      id: '2122',
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121',
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121',
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121',
    },
  ]

  it('should render default AtImagePicker', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- class', () => {
    const wrapper = factory({ class: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- files', () => {
    const wrapper = factory({ files: files })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- length', () => {
    const wrapper = factory({ files: files, length: 5 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- mode', () => {
    const wrapper = factory({ files: files, mode: 'top' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- showAddBtn', () => {
    const wrapper = factory({ files: files, showAddBtn: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- multiple', () => {
    const wrapper = factory({ files: files, multiple: true })
    expect(wrapper.element).toMatchSnapshot()
  })
})
