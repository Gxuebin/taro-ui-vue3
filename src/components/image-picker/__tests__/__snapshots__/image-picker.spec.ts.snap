// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtImagePicker should render default AtImagePicker 1`] = `
<view
  class="at-image-picker"
>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
  </view>
</view>
`;

exports[`AtImagePicker should render prop -- class 1`] = `
<view
  class="test at-image-picker test"
>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
  </view>
</view>
`;

exports[`AtImagePicker should render prop -- files 1`] = `
<view
  class="at-image-picker"
>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
  </view>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
  </view>
</view>
`;

exports[`AtImagePicker should render prop -- length 1`] = `
<view
  class="at-image-picker"
>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
  </view>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
  </view>
</view>
`;

exports[`AtImagePicker should render prop -- mode 1`] = `
<view
  class="at-image-picker"
>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="top"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="top"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="top"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="top"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
  </view>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="top"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
  </view>
</view>
`;

exports[`AtImagePicker should render prop -- multiple 1`] = `
<view
  class="at-image-picker"
>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
  </view>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
  </view>
</view>
`;

exports[`AtImagePicker should render prop -- showAddBtn 1`] = `
<view
  class="at-image-picker"
>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
  </view>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
  </view>
</view>
`;

exports[`AtImagePicker should render prop -- style 1`] = `
<view
  class="at-image-picker"
  style="color: red;"
>
  <view
    class="at-image-picker__flex-box"
  >
    <view
      class="at-image-picker__flex-item"
    >
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!---->
    </view>
  </view>
</view>
`;
